### ISSUE TITLE
Payload validation

### ISSUE THREAD

**Issue by alex10fan** [CircleOfNice/CiRestClientBundle#35]
CURLOPT_POSTFIELDS by default accepts strings and flat (1-dimensional) arrays.

Update payload validation of Curl->sendRequest method to allow arrays or let user to set payload manually using curl options (do not set payload if it is empty).

Without that it's impossible to post files.

$payload['file'] = new \CURLFile($file, $mimeType);
curl_setopt($ch, CURLOPT_POSTFIELDS, $payload);

**Comment by TobiasHauck** [comment]
Hey,
thanks for reporting and sorry for being so late :S 

There's another issue about file upload #36, so I think these issues are related to each other. Would you mind to find a solution? Maybe you can have a short chat with the other dev having the same problem, so you won't do the same in parallel.
